src/db/ConnectionPool.ts

```diff
@@ -96,6 +96,11 @@ const useTimeHistogram = new Histogram({
   labelNames: ['poolName', 'readonly'],
   buckets: [0.003, 0.005, 0.01, 0.05, 0.1, 0.3],
 });
+const maxConnectionLimitGauge = new Gauge({
+  name: 'db_pool_total_connections_limit_gauge',
```

Comment:
Being a bit picky, but could we call this `db_pool_max_pool_size_gauge`. I think it's clearer as it says that the pool will not grow over this limit.
But well... I'm happy either way